controllers/operatorpolicy_controller.go

```diff
@@ -727,6 +727,43 @@ func messageIncludesSubscription(subscription *operatorv1alpha1.Subscription, me
 	return regexp.MatchString(regex, message)
 }
 
+// constraintMessageMatch checks if the ConstraintsNotSatisfiable message is actually different
+// from the old one by sorting the clauses of the message
+func constraintMessageMatch(policy *policyv1beta1.OperatorPolicy, cond *operatorv1alpha1.SubscriptionCondition) bool {
+	const cnfPrefix = "constraints not satisfiable: "
+
+	var policyMessage, subMessage string
+
+	for _, cond := range policy.Status.Conditions {
+		if strings.Contains(cond.Message, cnfPrefix) {
+			policyMessage = cond.Message
+		}
+	}
+
+	if policyMessage == "" {
+		return false
+	}
+
+	if strings.Contains(cond.Message, cnfPrefix) {
+		subMessage = cond.Message
+	} else {
+		return false
+	}
+
+	policyMessage = strings.TrimPrefix(policyMessage, cnfPrefix)
+	subMessage = strings.TrimPrefix(subMessage, cnfPrefix)
```

Comment:
Nit, saves kind of saves a branch:
```suggestion
	if !strings.Contains(cond.Message, cnfPrefix) {
		return false
	}

	policyMessage = strings.TrimPrefix(policyMessage, cnfPrefix)
	subMessage = strings.TrimPrefix(cond.Message, cnfPrefix)
```